confmap/resolver_test.go

```diff
@@ -423,6 +423,120 @@ func TestResolverExpandMapAndSliceValues(t *testing.T) {
 	assert.Equal(t, expectedMap, cfgMap.ToStringMap())
 }
 
+func TestResolverExpandStringValues(t *testing.T) {
+	tests := []struct {
+		name   string
+		input  string
+		output any
+	}{
+		{
+			name:   "test_no_match_old",
+			input:  "${HOST}:${PORT}",
+			output: "${HOST}:${PORT}",
+		},
+		{
+			name:   "test_no_match_old_no_brackets",
+			input:  "${HOST}:$PORT",
+			output: "${HOST}:$PORT",
+		},
+		{
+			name:   "test_match_value",
+			input:  "${env:COMPLEX_VALUE}",
+			output: []any{"localhost:3042"},
+		},
+		{
+			name:   "test_match_embedded",
+			input:  "${env:HOST}:3043",
+			output: "localhost:3043",
+		},
+		{
+			name:   "test_match_embedded_multi",
+			input:  "${env:HOST}:${env:PORT}",
+			output: "localhost:3044",
+		},
+		{
+			name:   "test_match_embedded_concat",
+			input:  "https://${env:HOST}:3045",
+			output: "https://localhost:3045",
+		},
+		{
+			name:   "test_match_embedded_new_and_old",
+			input:  "${env:HOST}:${PORT}",
+			output: "localhost:${PORT}",
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			provider := newFakeProvider("input", func(context.Context, string, WatcherFunc) (*Retrieved, error) {
+				return NewRetrieved(map[string]any{tt.name: tt.input})
+			})
+
+			testProvider := newFakeProvider("env", func(_ context.Context, uri string, _ WatcherFunc) (*Retrieved, error) {
+				switch uri {
+				case "env:COMPLEX_VALUE":
+					return NewRetrieved([]any{"localhost:3042"})
+				case "env:HOST":
+					return NewRetrieved("localhost")
+				case "env:PORT":
+					return NewRetrieved("3044")
```

Comment:
I don't think this is what we get from `${env:HOST}`. See https://go.dev/play/p/0mEbAsegFKe